packages/algob/test/builtin-tasks/compile.ts

```diff
@@ -125,3 +125,71 @@ describe("Compile task", () => {
     assert.deepEqual(YAML.stringify(result), expected);
   });
 });
+
+describe("Support External Parameters in PyTEAL program", () => {
+  useFixtureProjectCopy("support-external-parameters");
+  const fakeAlgod: Algodv2 = {} as Algodv2; // eslint-disable-line @typescript-eslint/consistent-type-assertions
+  const op = new CompileOpMock(fakeAlgod);
+  const pyOp = new PyCompileOp(op);
+
+  const cacheDir = path.join("artifacts", "cache");
+  const stateless = "stateless.py";
+  const stateful = "stateful.py";
+  let statelessHash = 0;
+  let statefulHash = 0;
+
+  const scInitParam = {
+    TMPL_SENDER: 'KFMPC5QWM3SC54X7UWUW6OSDOIT3H3YA5UOCUAE2ABERXYSKZS5Q3X5IZY',
+    ASSET_AMOUNT: 1000
+  };
+
+  it("On first run, should compile pyTEAL code", async () => {
+    let result = await pyOp.ensureCompiled(stateless);
+    statelessHash = result.srcHash;
+
+    result = await pyOp.ensureCompiled(stateful);
+    statefulHash = result.srcHash;
+
+    const writtenFiles = [];
+    for (const fn of [stateless, stateful]) {
+      const fullF = path.join(cacheDir, fn + ".yaml");
+      writtenFiles.push(fullF);
+      assert.isTrue(fs.existsSync(fullF));
+    }
+
+    assert.equal(op.timestamp, 2);
+    assert.deepEqual(op.writtenFiles, writtenFiles);
+  });
+
+  it("On second run, shouln't recompile because No external parameters passed", async () => {
+    await op.resetAndCompile(false);
+
+    // timestamp didn't change because file is not recompiled
+    assert.equal(op.timestamp, 2);
+    assert.lengthOf(op.compiledFiles, 0);
+    assert.lengthOf(op.writtenFiles, 0);
+  });
+
+  it("On third run, should compile on third run because external parameters passed", async () => {
```

Comment:
will the last 2 unit tests work if the first one will fail? If not then we can put all of them in a single test (`it(...)`)

Reply:
ok